src/models/discriminant_analysis.jl

```diff
@@ -3,130 +3,123 @@
+  cov_b::CovarianceEstimator = MS.SimpleCovariance()
+  outdim::Int = 0::(_ ≥ 0)
+  regcoef::Float64 = 1e-6::(_ ≥ 0)
+  dist::SemiMetric = SqEuclidean()
 end
 
 const ERR_LONE_TARGET_CLASS = ArgumentError(
-    "The number of unique classes in "*
-    "the training target has to be greater than one, even if "*
-    "the complete pool contains more than one class. If "*
-    "`cov_b=CovarianceEstimation.SimpleCovariance()` this "*
-    "restriction on classes does not apply. "
+  "The number of unique classes in "*
+  "the training target has to be greater than one, even if "*
+  "the complete pool contains more than one class. If "*
+  "`cov_b=CovarianceEstimation.SimpleCovariance()` this "*
+  "restriction on classes does not apply. "
 )
 
 function _check_lda_data(model, X, y)
-    class_list = MMI.classes(y[1]) # Class list containing entries in pool of y.
-    nclasses = length(class_list)
-
-    # Class list containing entries in seen in y.
-    classes_seen = filter(in(y), class_list)
-    nc = length(classes_seen) # Number of classes in pool of y.
-    integers_seen = MMI.int(classes_seen)
-    Xm_t = _matrix_transpose(model, X) # Now p x n matrix
-    yplain = MMI.int(y) # Vector of n ints in {1,..., nclasses}.
-    p, n = size(Xm_t)
-
-    # Recode yplain to be in {1,..., nc}
-    nc == nclasses || _replace!(yplain, integers_seen, 1:nc)
-
-    # Check to make sure we have more than one class in training sample.
-    # This is to prevent Sb from being a zero matrix
-    # see issue #41
-    lone_class_unsupported = model isa Union{LDA, BayesianLDA} &&
-          model.cov_b != MS.SimpleCovariance()
-    if nc <= 1 && lone_class_unsupported
-        throw(ERR_LONE_TARGET_CLASS)
-    end
-
-    # Check to make sure we have more samples than classes.
-    # This is to prevent Sw from being the zero matrix.
-    if n <= nc
-        throw(
-            ArgumentError(
-                "The number of training samples `n` has"*
-                " to be greater than number of unique classes `nc`."
-            )
-        )
-    end
-    # Check output dimension default is min(p, nc-1)
-    def_outdim = min(p, nc - 1)
-    # If unset (0) use the default; otherwise try to use the provided one
-    outdim = ifelse(model.outdim == 0, def_outdim, model.outdim)
-    # Check if the given one is sensible
-    if outdim > def_outdim
-        throw(
-            ArgumentError(
-                "`outdim` must not be larger than `min(p, nc-1)`"*
-                "where `p` is the number of features in `X` and"*
-                "`nc` is the number of unique classes in the target vector."
-            )
-        )
-    end
-    return Xm_t, yplain, classes_seen, p, n, nc, nclasses, integers_seen, outdim
+  pool = MMI.classes(y[1]) # Class list containing entries in pool of y.
+  classes_seen = unique(y) # Class list of actual entries in seen in y.
+  nc = length(classes_seen) # Number of classes in pool of y.
```

Comment:
The comment here is confusing. Isn't the point that the `nc` is possibly *different* from the number of classes in the pool? This is number of classes seen, not the total number of classes in the pool, right?

Reply:
Yes, `nc` is the total number of classes seen (i.e actual number of classes present in the training data) which may be different from the total  no. of classes in the pool of y. I would fix that comment as it's indeed confusing.